src/main/java/gov/usgs/earthquake/indexer/DefaultIndexerModule.java

```diff
@@ -125,6 +84,7 @@ public ProductSummary getProductSummary(final Product product)
 	 * @return the absolute preferred weight.
 	 */
 	protected long getPreferredWeight(final ProductSummary summary) {
+		Regions regions = ANSSRegionsFactory.getFactory().getRegions();
```

Comment:
Any timing concerns here? Is it possible for regions to be null on slow connections or does this just block?

Reply:
This is synchronous.